File: Maximum-Distance-Between-a-Pair-of-Values.py
1class Solution:
2    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
3        left,right = 0,0
4        maxdist = -1
5        while(left<len(nums1) and right<len(nums2)):
6            if(left<=right and nums1[left]<=nums2[right]):
7                maxdist = max(maxdist,right-left)
8                right+=1
9            else:
10                left+=1
11                right+=1
12        return maxdist if maxdist!=-1 else 0

File: Maximum-Distance-Between-a-Pair-of-Values.cpp
1class Solution {
2public:
3    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
4        int left = 0, right = 0;
5        int maxdist = INT_MIN;
6        while(left < nums1.size() && right < nums2.size()){
7            if(left<=right){
8                if(nums1[left] <= nums2[right]){
9                    maxdist = max(maxdist,right-left);
10                    right++;
11                }
12                else{
13                    left++;
14                }
15            }
16            else{
17                right = left;
18            }
19        }
20        return (maxdist!=INT_MIN) ? maxdist : 0;
21    }
22};